src/main/java/jenkins/plugins/slack/StandardSlackService.java

```diff
@@ -142,16 +142,81 @@ void correctMisconfigurationOfBaseUrl() {
         }
     }
 
+    /**
+     * Make an HTTP POST to the Slack API
+     *
+     * @param apiEndpoint - The API endpoint to request, e.g. `chat.postMessage`
+     * @param body - The payload body to be POSTed to the API
+     *
+     * @return boolean indicating whether the API request succeeded
+     */
+    boolean makeApiPost(String apiEndpoint, JSONObject body) {
+        boolean result = true;
+
+        if (baseUrl != null) {
+            correctMisconfigurationOfBaseUrl();
+        }
+
+        try (CloseableHttpClient client = getHttpClient()) {
+            HttpPost post;
+            String url;
+
+            //prepare post methods for both requests types
+            if (!botUser || StringUtils.isNotEmpty(baseUrl)) {
+                // TODO how does this work for non-chat.postMessage requests?
```

Comment:
The content here is pasted verbatim from what was in `publish(SlackRequest)`, except that I've added this comment. Any concerns about this code path, which doesn't use `apiEndpoint`?